### ISSUE TITLE
CI checks and/or action registration should not allow invalid type

### ISSUE THREAD

**Issue by LindsayHill** [StackStorm/st2contrib#531]
If I create an action metadata file containing a section with an invalid parameter type, CI checks will pass, and I can register the action. It only fails when I attempt to use it.

Example YAML snippet:
```
  commands:
    type: "list"
    description: "Configuration commands to run"
    required: true
```
This action can be registered, and will show up if I run `st2 run packname.action -h`, e.g.:
```
$ st2 run clicrud.configure -h|grep -C 2 list
    commands
        Configuration commands to run
        Type: list

    host
```
But if I attempt to use that action, it (correctly) fails with a 500 error:
```
ERROR: 500 Server Error: Internal Server Error
MESSAGE: u'list' is not valid under any of the given schemas

Failed validating 'anyOf' in schema['properties']['properties']['additionalProperties']['properties']['type']:
    {'anyOf': [{'$ref': '#/definitions/simpleTypes'},
               {'items': {'$ref': '#/definitions/simpleTypes'},
                'minItems': 1,
                'type': 'array',
                'uniqueItems': True}]}

On instance['properties'][u'commands']['type']:
    u'list' for url: http://127.0.0.1:9101/v1/executions
```

I believe that we should identify this misconfiguration sooner in the pipeline, rather than at action run time.

**Comment by LindsayHill** [comment]
Being addressed - see commentary in #532

**Comment by Kami** [comment]
Fixed by https://github.com/StackStorm/st2/pull/2872 - thanks!
